File: FMDB的使用/FMDB的使用/ViewController.m
//
//  ViewController.m
//  FMDB的使用
//
//  Created by Apple on 16/8/6.
//  Copyright © 2016年 Apple. All rights reserved.
//

#import "ViewController.h"
#import "FMDB.h"
#import <sqlite3.h>
#import <sqlite3_private.h>
#import <sqlite3ext.h>
#define FMDBQuickCheck(SomeBool) { if (!(SomeBool)) { NSLog(@"Failure on line %d", __LINE__); abort(); } }
@interface ViewController ()
@property(nonatomic,strong)FMDatabase*dataBase;
@end
@implementation ViewController
- (void)viewDidLoad {
    [super viewDidLoad];
    //FMDB 是使用  增删改查
    //获取路径数组
    NSArray *paths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
    //获取路径
    NSString *documentDirectory = [paths objectAtIndex:0];
    //dbPath： 数据库路径，在Document中。
    NSString *dbPath = [documentDirectory stringByAppendingPathComponent:@"Test.db"];
    NSLog(@"%@",dbPath);
    //创建db对象
    FMDatabase *db= [FMDatabase databaseWithPath:dbPath] ;
    if (![db open]) {
        NSLog(@"Could not open db.");
        return ;
    }
    //创建表和类型
    [db executeUpdate:@"CREATE TABLE  IF NOT EXISTS User (id integer PRIMARY KEY autoincrement , Name text, Age integer)"];
//    NSLog(@"%@",[db lastError]);
//   NSLog(@"%@",[db lastErrorMessage]);
    //增加数据
    [db executeUpdate:@"INSERT INTO User (Name,Age) VALUES (?,?)",@"老婆",[NSNumber numberWithInt:20]] ;
    //更改数据
    [db executeUpdate:@"UPDATE User SET Name = ? WHERE Name = ? ",@"宝贝",@"老婆"];
    //删除数据
//    [db executeUpdate:@"DELETE FROM User WHERE Name = ?",@"宝贝"];
    //查询数据  //只能查询一条
    NSString *aa=[db stringForQuery:@"SELECT Name FROM User WHERE Age = ?",@"20"];
    NSLog(@"%@",aa);
    //查询所有符合的数据
   FMResultSet*rs=[db executeQuery:@"SELECT * FROM User WHERE Age = ?",@"20"];
    while ([rs next])//指向下一个,判断是否有值
    {
        NSLog(@"%@ %@",[rs stringForColumn:@"Name"],[rs stringForColumn:@"Age"]);
    }
    //批量创建数据
    BOOL oldshouldcachestatements = db.shouldCacheStatements;
    [db setShouldCacheStatements:YES];//开启缓存
    [db beginTransaction];//开启事务
    //在nsstring %@ 起作用,可以替换表名
    NSString *insertQuery = [[NSString alloc] initWithFormat:@"INSERT INTO User values( null,?, ? );"];
    BOOL success = false;//定义属性接受是否写入成功
   
        for (int record = 0; record < 100; record++) {
             @autoreleasepool {
            //在sqlite中?起作用,替换字段值
            success = [db executeUpdate:insertQuery,[NSString stringWithFormat:@"小老婆%d",record],[NSNumber numberWithInt:18]];
        }
//        NSLog(@"%d",success);
        [db commit];//提交事务
        [db setShouldCacheStatements:oldshouldcachestatements];//不开启缓存
            [db close];//关闭数据库
//        NSLog(@"%@",[db lastError]);
//        NSLog(@"%@",[db lastErrorMessage]);
    }
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: 微信支付/swift使用oc/oc中使用swift/swift基础/swift基础/swift基础/ViewController.swift
//
//  ViewController.swift
//  swift基础
//
//  Created by Apple on 16/10/12.
//  Copyright © 2016年 lkb-求工作qq:1218773641. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    var finish :(() -> ())?
    var s : Int?
//    ={() -> () in
//    print("21213")
//    }
    override func viewDidLoad() {
        super.viewDidLoad()
//        demo4(2,b: 1,c: 4)
        demo5 { () -> () in
            print("我来了,闭包")
        }
    }
    //闭包//没有名字,没有参数
    func demo5(finished: ()->()){
    
//    finished()
         { () -> () in
            print("我来了,闭包")
        }()

    
    }
    private func demo4(a:Int,b:Int,c:Int) ->Int{
    
    
    
    return a + b
    }
    private func demo3() {
        var dict = ["name":"李康滨","age":18]
        print(dict)
        dict["title"]="女神"
        print(dict)
        dict["name"]="帅哥"
         print(dict)
        dict.removeValueForKey("name")
          print(dict)
        for(key,vlaue) in dict {
        print("key=\(key)")
             print("vlaue=\(vlaue)")
        
        }
        
    
    
    }
    private func demo2(){
    
    let arr1 = [1,2,3]
        let arr2 = [4,5]
        print(arr1+arr2)
    
    
    }
    private func demo1(){
    var array = [1,3,2,343,"李冰冰"]
        for  i  in array {
        print(i)
        
        
        }
        array.append("范冰冰")
        print(array)
        array[0] = "芬姐"
        print(array)
       let ss = array.dropFirst()
        print(ss)
        array.removeFirst()
        print(array)
        

        
    
    
    
    
    
    
    
    }
    private func demo(){
    //截取字符串
     let str = "我是字符串,截取,我过来了"
//        let nsstr = (str as NSString).substringWithRange(NSMakeRange(2, 4));
        let s = str.startIndex.advancedBy(2)
        let end = str.endIndex.advancedBy(-1)
        let ss = str.substringWithRange(s..<end)
        
        print(ss)
        
        
    
    
    }
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

